packages/orbit-components/src/tmp-tailwind/Textarea/index.tsx

```diff
@@ -0,0 +1,147 @@
+import FormLabel from "../../FormLabel";
+import { SIZE_OPTIONS, RESIZE_OPTIONS } from "../../Textarea/consts";
+import useErrorTooltip from "../../ErrorFormTooltip/hooks/useErrorTooltip";
+import getFieldDataState from "../../common/getFieldDataState";
+import type { Props } from "../../Textarea/types";
+import useRandomId from "../../hooks/useRandomId";
+import { getSpaceAfterClasses } from "../../common/tailwind";
+
+const Textarea = React.forwardRef<HTMLTextAreaElement, Props>((props, ref) => {
+  const {
+    size = SIZE_OPTIONS.NORMAL,
+    disabled,
+    resize = RESIZE_OPTIONS.VERTICAL,
+    dataTest,
+    id,
+    spaceAfter,
+    fullHeight,
+    defaultValue,
+    value,
+    label,
+    name,
+    error,
+    placeholder,
+    helpClosable = true,
+    maxLength,
+    onChange,
+    onFocus,
+    onBlur,
+    tabIndex,
+    help,
+    rows,
+    readOnly,
+    required,
+    dataAttrs,
+  }: Props = props;
+
+  const forID = useRandomId();
+  const inputId = id || forID;
+
+  const hasTooltip = Boolean(error || help);
+
+  const {
+    tooltipShown,
+    tooltipShownHover,
+    setTooltipShown,
+    setTooltipShownHover,
+    labelRef,
+    iconRef,
+    handleFocus,
+  } = useErrorTooltip({ onFocus, hasTooltip });
+
+  const shown = tooltipShown || tooltipShownHover;
+
+  return (
+    <label
+      className={cx(
+        "font-base relative flex w-full flex-col",
+        fullHeight && "h-full flex-1",
+        spaceAfter && getSpaceAfterClasses(spaceAfter),
+      )}
+      ref={labelRef}
+    >
+      {label && (
+        <FormLabel
+          filled={!!value}
+          error={!!error}
+          help={!!help}
+          required={required}
+          disabled={disabled}
+          iconRef={iconRef}
+          onMouseEnter={() => setTooltipShownHover(true)}
+          onMouseLeave={() => setTooltipShownHover(false)}
+        >
+          {label}
+        </FormLabel>
+      )}
+
+      <textarea
+        className={cx(
+          "w-full appearance-none",
+          "font-base",
+          "rounded-normal relative box-border block overflow-auto",
+          "min-h-form-box-normal",
+          "duration-fast transition-shadow ease-in-out",
+          "border border-transparent",
+          // TODO: remove when will be migrated from tmp-folder
+          size === SIZE_OPTIONS.SMALL && "text-normal px-sm py-xs leading-normal",
+          resize === RESIZE_OPTIONS.VERTICAL ? "resize-y" : "resize-none",
+          size === SIZE_OPTIONS.NORMAL && "text-normal p-sm leading-normal",
+          disabled ? "cursor-not-allowed" : "cursor-text",
+          fullHeight && "h-full flex-1",
+          disabled || readOnly
+            ? "bg-form-element-disabled-background"
+            : [
+                "bg-form-element-background",
+                error
+                  ? "shadow-form-element-error hover:shadow-form-element-error-hover"
+                  : "shadow-form-element hover:shadow-form-element-hover",
```

Comment:
This can't be like this. This is saying the error shadow only appears when it's not disabled nor readOnly, which is not true. Only hover shadow is affected disabled || readOnly.

The suggestion [here](https://github.com/kiwicom/orbit/pull/4053#discussion_r1345877876) has it correctly. It adds the the shadow depending on error state (line 107) and then the hover shadow depends on the disabled || readOnly (line 114).

Reply:
it can be disabled and have the error shadow?